src/pushNotifications/widget/pushNotifications.js

```diff
@@ -297,17 +297,41 @@ define([
                 deviceRegistration.set("DeviceType", "iOS");
             }
 
-            // We commit the DeviceRegistration object to trigger the backend process.
-            // This is the only time we can commit the object, because it will be deleted in the AfterCommit event.
-            mx.data.commit({
-                mxobj: deviceRegistration,
-                callback: dojoLang.hitch(this, function() {
-                    logger.debug("Registered device with ID " + deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
-                }),
-                error: function(e) {
-                    logger.error("Error occurred attempting to register device: " + e);
-                }
-            });
+            // CC Jan 10, 2018 - only commit the DeviceRegistration if the registration id has changed or does not yet exist
+            var lastRegistrationId = localStorage.getItem("mx-registration-id");
+            if (!lastRegistrationId || lastRegistrationId !== deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE)) {
+                // We commit the DeviceRegistration object to trigger the backend process.
+                // This is the only time we can commit the object, because it will be deleted in the AfterCommit event.
+                mx.data.commit({
+                    mxobj: deviceRegistration,
+                    callback: dojoLang.hitch(this, function() {
+                        logger.debug("Registered device with ID " + deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
+                        if (mx.isOffline()) {
+                            mx.data.synchronizeOffline({ // settings, success, error
+                                    fast: true // data only
+                                },
+                                dojoLang.hitch(this, function() {
+                                    console.log("Synchronization successful.")
+                                    if (window.localStorage) {
+                                        console.debug("writing new device registration id to localstorage: " + deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
+                                        localStorage.setItem("mx-registration-id", deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
```

Comment:
You only store it in localStorage when in offline mode. If we store it always after successful commit/synchronization, then this optimization would work in online mode as well.

Reply:
when online, the commit to the mx database already covers this functionality, so my thought was that this is only necessary for offline apps